File: programmers/lv0/코딩기초트레이닝/연산과조건문/조건문자열/Solution.js
// const solution = (ineq, eq, n, m) => {
//     let answer = false
//     switch (ineq + eq) {
//         case '<=': answer = n <= m; break
//         case '<!': answer = n < m; break
//         case '>=': answer = n >= m; break
//         case '>!': answer = n > m; break
//     }
//     return answer ? 1 : 0
// }

// simple
const solution = (ineq, eq, n, m) => {
    const ops = {
        '>=': (n, m) => n >= m,
        '<=': (n, m) => n <= m,
        '>!': (n, m) => n > m,
        '<!': (n, m) => n < m,
    }
    const op = ops[ineq + eq]
    return Number(op(n, m))
}

console.log(solution('<', '=', 20, 50))
console.log(solution(">", "!", 41, 78))

File: programmers/lv0/코딩기초트레이닝/연산과조건문/조건문자열/Solution.swift
import Foundation

func solution(_ ineq:String, _ eq:String, _ n:Int, _ m:Int) -> Int {
    let expression = ineq + eq
    if (expression == "<=") {
        return n <= m ? 1 : 0
    }
    else if (expression == "<!") {
        return n < m ? 1 : 0
    }
    else if (expression == ">=") {
        return n >= m ? 1 : 0
    }
    else {
        return n > m ? 1: 0
    }
}

print(solution("<", "=", 20, 50))
print(solution(">", "!", 41, 78))
